products/BellHybrid/services/db/databases/MeditationStatisticsTable.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "MeditationStatisticsTable.hpp"
+#include <time.h>
+#include <chrono>
+#include <date/date.h>
+
+namespace
+{
+    using namespace db::meditation_stats;
+    constexpr auto tableName = "meditation_stats";
+
+    /// We could use the SQLite function "datetime(timestamp,'unixepoch') to generate the correct time string.
+    /// Unfortunately, SQLite v3.38.1 has some problems converting UNIX timestamps. That's why we decided we would store
+    /// timestamps using string format, as it doesn't cause any issues.
+
+    std::string prepare_timestamp(time_t unix_timestamp)
+    {
+        char time_buf[64]{};
+        struct tm ts
+        {};
+        ts = *gmtime(&unix_timestamp);
+        strftime(time_buf, sizeof(time_buf), "%Y-%m-%d %H:%M:%S", &ts);
+
+        return time_buf;
+    }
+
+    time_t to_unixepoch_timestamp(std::string_view str)
+    {
+        std::istringstream in{str.data()};
+        date::sys_seconds tp;
+        in >> date::parse("%Y-%m-%d %H:%M:%S", tp);
+        return tp.time_since_epoch().count();
+    }
+
+    std::vector<TableRow> get_by_x(Database *db, const std::string_view modifier, const std::uint32_t x)
+    {
+        if (x == 0) {
+            return {};
+        }
+
+        const auto retQuery = db->query("SELECT * from %s where timestamp BETWEEN "
+                                        "datetime('now','-%lu %s') and datetime('now');",
+                                        tableName,
+                                        x,
+                                        modifier.data());
+
+        if ((retQuery == nullptr) || (retQuery->getRowCount() == 0)) {
+            return {};
+        }
+
+        std::vector<TableRow> ret;
+        ret.reserve(retQuery->getRowCount());
+
+        do {
+            ret.push_back(TableRow{
+                {(*retQuery)[0].getUInt32()},                       // ID
+                to_unixepoch_timestamp((*retQuery)[1].getString()), // timestamp
+                std::chrono::minutes{(*retQuery)[2].getUInt32()}    // duration
+            });
+        } while (retQuery->nextRow());
+
+        return ret;
+    }
+} // namespace
+
+namespace db::meditation_stats
+{
+    MeditationStatsTable::MeditationStatsTable(Database *db) : Table(db)
+    {}
+
+    auto MeditationStatsTable::create() -> bool
+    {
+        return true;
+    }
+    auto MeditationStatsTable::removeById([[maybe_unused]] uint32_t id) -> bool
```

Comment:
What about `std::uint32_t`

Reply:
These are the implementations of the "module-db/Tables/Table.hpp" interface :)